Design.md

```diff
@@ -227,8 +229,23 @@ This API will enable or disable user on Vault.
 
 
 ### Create S3 Credential
-This API creates S3 credential for the user.
-1. `generate-access-key` api will be called using assumed role credentials.
+This API creates S3 credentials for the user.
+1. The Generate Access Key API will be called using assumed role credentials.
+1. Encrypt the secret key (For more, see [SecretKey-Encryption-Strategy](#SecretKey-Encryption-Strategy)). Store the encrypted secret key on Redis Sentinel in the hash named "osis:s3credentials", with the hash key formatted as `<Username>__<AccessKeyID>`.
+1. Log the access key ID _(Log location and tools for logging are yet to be decided)_
+
+**Important Notes:**
+* Redis Sentinel is subject to an ongoing rolling deployment.
+* If Redis Sentinel fails or crashes, the vCloud Director's OSE service must be restarted with the following command:_
+    ```shell
+    $ ose service restart
+    ```
+  * All access keys created by OSIS before OSE restart time (`ose-restart-time`) are dead keys on the Vault DB, which OSE/OSIS cannot use and must be manually deleted from the Vault. 
```

Comment:
What about any workloads running (while OSE is restarted) and using this keypair for authentication? Would an OSE service restart require these applications to get new credentials somehow (and hence potentially restart) as well?

Reply:
Addressed in 1779625549f7a9e63533d01d48aa33d3f491abb4

> Once OSE service restart process is finished,
     * OSE will invoke `listS3Credentials` API to identify the keys for object storage operations. OSIS will return only an empty list for `listS3Credentials` API (as no key  is available for the user on Redis).
     * If no keys are present for a user, OSE will invoke `createS3Credentials` API to create a new key for object storage operations.